pyspider/scheduler/scheduler.py

```diff
@@ -574,7 +574,7 @@ def on_old_request(self, task, old_task):
         schedule_age = _schedule.get('age', self.default_schedule['age'])
         if _schedule.get('itag') and _schedule['itag'] != old_schedule.get('itag'):
             restart = True
-        elif schedule_age >= 0 and schedule_age + (old_task['lastcrawltime'] or 0) < now:
+        elif 'lastcrawltime' in old_task and schedule_age >= 0 and schedule_age + (old_task['lastcrawltime'] or 0) < now:
```

Comment:
`elif schedule_age >= 0 and schedule_age + old_task.get('lastcrawltime', 0) < now:`
could be better.